### PR TITLE
Don't redefine Corrosion::Generator if find_package is used multiple times

### PR DESCRIPTION
**Pull request by ratmice**
Here was my blind attempt at fixing #142 which may/may not be the right way to fix it

### CODE REVIEW

File: cmake/CorrosionConfig.cmake.in [hunk 1]
```diff
@@ -6,7 +6,7 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH "${PACKAGE_PREFIX_DIR}/@CORROSION_INSTALL_PREFIX@@CMAKE_INSTALL_DATADIR@/cmake")
 
-add_executable(Corrosion::Generator IMPORTED GLOBAL)
+add_executable(Corrosion::Generator IMPORTED)
```
Comment (tronical) [inline cmake/CorrosionConfig.cmake.in:9]: I think the idiomatic way is typically to keep the executable as a global target (why localise it?), but guard it with an `if (NOT TARGET Corrosion::Generator)`, like in commit 46d715d64165037cb740181f379088a0f077dbba.

That would span all the way until line 19/20 I think.

Comment (hunger) [reply cmake/CorrosionConfig.cmake.in:9]: Without the GLOBAL this target has directory scope. I doubt that is enough: You probably do want to use the Corrosion::Generator somewhere outside of this directory (and maybe below).

Comment (ratmice) [reply cmake/CorrosionConfig.cmake.in]: @tronical seems like a much better fix which works for me, thank you.

**Comment by tronical** [comment]
👍

**Comment by ogoffart** [comment]
Thanks for the patch!

**Pull request merged**